modules/apps/29-fee/ibc_module.go

```diff
@@ -175,9 +175,17 @@ func (im IBCModule) OnRecvPacket(
 	ctx sdk.Context,
 	packet channeltypes.Packet,
 	relayer sdk.AccAddress,
-) ibcexported.Acknowledgement {
-	// TODO: Implement fee specific logic if fee is enabled for the given channel
-	return im.app.OnRecvPacket(ctx, packet, relayer)
+) exported.Acknowledgement {
+	ack := im.app.OnRecvPacket(ctx, packet, relayer)
+	sourceRelayer, found := im.keeper.GetCounterpartyAddress(ctx, relayer.String())
```

Comment:
This needs IsFeeEnabled check. If the fee is not enabled for this channel, the acknowledgement should not be wrapped